src/main/java/org/linksharing/server/http/ApiRestController.java

```diff
@@ -66,18 +70,39 @@ ResponseEntity<UserProfileDetails> updateProfileDetails(Principal user, @Request
     }
 
     @GetMapping("/profile/picture")
-    ResponseEntity<?> getProfilePicture(Principal user) throws FileNotFoundException {
+    ResponseEntity<?> getProfilePicture(Principal user) throws IOException {
 
-        File img = new File("../image.jpg");
+        UserProfileDetails userProfile = profileRepository.findByEmail(user.getName());
+        String imageUrl;
+
+        if (userProfile.getImageUrl() != null) {
+            imageUrl = userProfile.getImageUrl();
+        } else  {
+            imageUrl = "src/main/resources/static/placeholder.jpg";
+        }
+
+        File img = new File(imageUrl);
         return ResponseEntity
                 .ok()
                 .contentType(IMAGE_JPEG)
                 .body(new InputStreamResource(new FileInputStream(img)));
     }
 
     @PostMapping("/profile/picture")
-    ResponseEntity<?> updateProfilePicture(Principal user) {
-        return null;
+    ResponseEntity<UserProfileDetails> updateProfilePicture(Principal user, @RequestParam("file") MultipartFile file) throws IOException {
+        UserProfileDetails userProfile = profileRepository.findByEmail(user.getName());
+
+        String[] fileNameParts = file.getOriginalFilename().split("\\.");
+        String fileExtension = "." + fileNameParts[fileNameParts.length - 1];
+
+        Path newFileName = Paths.get("src/main/resources/upload/", userProfile.getEmail() + fileExtension);
```

Comment:
Choose some folder which is in the same folder as the working dir (working directory is the directory where the app in running). 
If you're running the app from Idea the working dir is equal to project dir. 
Don't forget to exclude the folder from Git. We don't need user data in here.

Reply:
To go after New Year)))

Reply:
You mean that the "pics" folder should be just under the working directory?
Like this?
MyProjects -> link-sharing-app -> pics

Or next to working directory, like this?
MyProjects -> link-sharing-app
MyProjects -> pics
MyProjects -> some-other-project